lib/inspec/cli.rb

```diff
@@ -93,6 +93,13 @@ def check(path) # rubocop:disable Metrics/AbcSize
     exit 1 unless result[:summary][:valid]
   end
 
+  desc 'vendor', 'Download all dependencies and generate a lockfile'
+  def vendor(path = nil)
+    profile = Inspec::Profile.for_target('./', opts)
+    lockfile = profile.generate_lockfile(path)
+    File.write('inspec.lock', lockfile.to_yaml)
```

Comment:
Currently `path` refers to the path to the vendored dependencies.  If we want we can make it refer to the path to the profile we want to lock and then make the path to the vendored dependencies be something you pass via a flag.